.circleci/config.yml

```diff
@@ -0,0 +1,120 @@
+version: 2.1
+
+orbs:
+  codecov: codecov/codecov@1.0.5
+
+executors:
+  my-executor:
+    docker:
+      - image: humancompatibleai/imitation:base
+    working_directory: /imitation
+    environment:
+      # If you change these, also change ci/code_checks.sh
+      SRC_FILES: src/ tests/ experiments/ setup.py
+
+commands:
+  dependencies:
+    # You must still manually update the Docker image if any
+    # binary (non-Python) dependencies change.
+    description: "Check out and update Python dependencies."
+    steps:
+      - checkout
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "setup.py" }}
+
+      - run:
+          name: install dependencies
+          # MUJOCO_KEY is defined in a CircleCI context
+          # Do some sanity checks to make sure key works
+          command: |
+            curl -o /root/.mujoco/mjkey.txt ${MUJOCO_KEY}
+            md5sum /root/.mujoco/mjkey.txt
+            # Only create venv if it's not been restored from cache
+            # TODO(adam): build_venv
+            [[ -d venv ]] || USE_MPI=True ./ci/build_venv.sh
+            python -c "import mujoco_py"
+
+      - save_cache:
+          paths:
+            - ./venv
+          key: v1-dependencies-{{ checksum "setup.py" }}
+
+      - run:
+          name: install imitation
+          # Build a wheel then install to avoid copying whole directory (pip issue #2195)
+          command: |
+            python setup.py sdist bdist_wheel
+            pip install --upgrade dist/imitation-*.whl
+
+jobs:
+  lintandtype:
+    executor: my-executor
+
+    steps:
+      - dependencies
+      - run:
+          name: flake8
+          command: flake8 ${SRC_FILES}
+
+      - run:
+          name: codespell
+          command: codespell -I .codespell.skip --skip='*.pyc,tests/data/*,*.ipynb,*.csv' ${SRC_FILES}
+
+      - run:
+          name: sphinx
+          command: pushd docs/ && make clean && make html && popd
+
+      - run:
+          name: pytype
+          command: pytype ${SRC_FILES}
```

Comment:
Why do we duplicate `ci/code_checks.sh` commands here instead of calling the script? (Maybe since this allows us check all of them even if the first check fails?)

Reply:
So we have a pretty GUI that gives a breakdown of each stage and timing for it.

If we don't care for that, it'd be more DRY to just call the script.

Reply:
Keeping pretty GUI seems ok for now. Don't think we will need to update code_checks.sh often.